src/main/java/io/cdap/plugin/http/source/common/pagination/IncrementAnIndexPaginationIterator.java

```diff
@@ -61,4 +65,28 @@ protected String getNextPageUrl(HttpResponse response, BasePage page) {
   public boolean supportsSkippingPages() {
     return true;
   }
+
+  @Override
+  public void setNextPageUrl(String nextPageUrl) {
+    super.setNextPageUrl(nextPageUrl);
+
+    Matcher matcher = getUrlPattern().matcher(nextPageUrl);
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException(String.format(
+        "Cannot set nextPageUrl to '%s' as it does not match the original url '%s'", nextPageUrl, config.getUrl()));
+    }
+
+    this.index = Long.parseLong(matcher.group(1));
+  }
+
+  private Pattern getUrlPattern() {
+    String[] urlParts = config.getUrl().split(PAGINATION_INDEX_PLACEHOLDER_REGEX);
+
+    String regexpString = Pattern.quote(urlParts[0]) + "([\\d]+)";
+    if (urlParts.length > 1) {
+      regexpString += Pattern.quote(urlParts[1]);
+    }
+
+    return Pattern.compile(regexpString);
```

Comment:
this operation is not cheap. Can it be compiled once?

Reply:
done